### PR TITLE
Register key with the store

### PR DESCRIPTION
**Pull request by slimjim777**
Register a signing key with the store for a sub-brand. This is the equivalent of a snap register-key.

### CODE REVIEW

File: store/store.go [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+	AuthorityID string `json:"authority-id"`
+	KeyName     string `json:"key-name"`
+}
+
+// RegisterKey registers an account-key assertion with the store
+func RegisterKey(keyAuth KeyRegister, keypair datastore.Keypair) error {
+
+	// Login to the store as the user
+	// Permissions are documented at https://dashboard.snapcraft.io/docs/api/macaroon.html#request-a-macaroon
+	permissions := []string{"edit_account", "modify_account_key"}
```
Comment (rmescandon) [inline store/store.go]: Any plan to handle these permissions from the UI or are they fix?

File: store/store.go [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+	_, err = client.Do(r)
+	if err != nil {
+		log.Printf("Error submitting the account-key assertion: %v", err)
+		return err
+	}
+
+	// defer resp.Body.Close()
+
+	// b, err := ioutil.ReadAll(resp.Body)
+	// log.Println(string(b))
```
Comment (rmescandon) [inline store/store.go]: Why this commented out code?

File: store/store.go [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+		"account_key_request": string(assert),
+	}
+	d, err := json.Marshal(data)
+	if err != nil {
+		log.Printf("Error marshalling account-key assertion: %v", err)
+		return err
+	}
+
+	// Submit the account-key assertion
+	r, _ := http.NewRequest("POST", storeBaseURL+"account/account-key", bytes.NewReader(d))
```
Comment (rmescandon) [inline store/store.go]: sounds to me this method to be in a separate place, like a generic http client

File: store/store.go [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+
+	perm := Permissions{Permissions: permissions}
+	macaroonJSONData, err := json.Marshal(perm)
+	if err != nil {
+		log.Println("Error marshalling the macaroon", err)
+		return "", err
+	}
+
+	// Submit the account-key assertion
+	r, err := http.Post(storeBaseURL+"acl/", "application/json", bytes.NewReader(macaroonJSONData))
```
Comment (rmescandon) [inline store/store.go]: same as before, maybe it would be nice having all http client operations in same file/place

File: store/store.go (hunk 1 above)
Comment (slimjim777) [reply store/store.go]: They are really fixed. These are the minimum that we need for the operations that we are using.

File: store/store.go (hunk 4 above)
Comment (slimjim777) [reply store/store.go]: I'll look at it, though the style of the two calls is different.

File: store/store.go (hunk 2 above)
Comment (slimjim777) [reply store/store.go]: That's leftover debugging - it was a long painful journey to get to this solution!

File: store/store.go (hunk 4 above)
Comment (slimjim777) [reply store/store.go]: I've done it, but it doesn't look great to me. They are different calls: one with custom headers and one without, one with a straight POST and the other with the response being parsed.

**Pull request merged**